app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -24,11 +24,18 @@ def plan(job_invocation, host, template_invocation, proxy)
 
         raise _('Could not use any template used in the job invocation') if template_invocation.blank?
 
-        settings = { :global_proxy   => 'remote_execution_global_proxy',
-                     :fallback_proxy => 'remote_execution_fallback_proxy' }
+        if proxy.blank?
+          offline_count = options[:offline_count] || 0
+          raise n_('The only usable proxy is down',
+                   'All %{number} usable proxies are down.',
+                   offline_count) % { :number => offline_count } if offline_count > 0
```

Comment:
Indeed.  I think the error only shows up in the task output, right? So I don't think it's such a big problem if it's huge, it's not being shown in a flash or something small.

But, you could limit it to showing the first 5, and then say something like "and 17 more."